test/adapter/json_api/relationship_test.rb

```diff
@@ -4,132 +4,204 @@ module ActiveModelSerializers
   module Adapter
     class JsonApi
       class RelationshipTest < ActiveSupport::TestCase
+        class Author < ActiveModelSerializers::Model
+          attr_accessor :id, :name, :blog, :posts
+        end
+        class AuthorSerializer < ActiveModel::Serializer
+          type :authors
+          attributes :id, :name
+
+          has_one :blog
+          has_one :blog_without_serializer
+          has_many :posts
+        end
+        class Blog < ActiveModelSerializers::Model
+          attr_accessor :id, :name
+        end
+        class BlogWithoutSerializer < Blog; end
+        class BlogSerializer < ActiveModel::Serializer
+          type :blogs
+          attributes :id, :name
+        end
+
         setup do
-          @blog = Blog.new(id: 1)
-          @author = Author.new(id: 1, name: 'Steve K.', blog: @blog)
-          @serializer = BlogSerializer.new(@blog)
-          ActionController::Base.cache_store.clear
+          @serializable_resource_options = {} # adapter.instance_options
+        end
+
+        def build_author_relationship(author_attributes, association_name)
```

Comment:
I completely rewrote the test harness and assertion in this file.. could use a second pair of eyes, but at least I think it is straightforward enough to read the assertions and understand, and to read the harness and understand what it's doing (even if not so clear why it's so complicated, and how could it be simpler).